src/player.cpp

```diff
@@ -4151,7 +4151,7 @@ void player::apply_damage(Creature *source, body_part hurt, int dam)
         hp_cur[hurtpart] = 0;
     }
 
-    if( hp_cur[hurtpart] <= 0 ) {
+    if( hp_cur[hurtpart] <= 0 && (source != nullptr && !source->is_hallucination())) {
```

Comment:
maybe just check if `dam > 0` instead?

Reply:
That fix has already been rejected as far as I'm aware

Reply:
What's the `source` when you fall in a pit or trip on rubble?  If it's `null`, then those would prevent mending from resetting

Reply:
Good point. I didn't see that in my review but it's likely to be null.

Looks like it'll just have to be some extra logic of that's the case.

Reply:
I think you can just do ```( source == nullptr || !source->is_hallucination() )```
That way damage from the environment will still reset mending.

Reply:
@kevingranade thank you... this really shouldn't have been so difficult for me -.-